internal/cli/server/config.go

```diff
@@ -810,7 +810,7 @@ func (c *Config) buildEth(stack *node.Node) (*ethconfig.Config, error) {
 	}
 
 	// snapshot disable check
-	if c.Snapshot {
+	if c.NoSnapshot {
```

Comment:
not sure but how come Snapshot and NoSnapshot checks are same

Reply:
Yes, so there was a bug in the new-cli implementation. It was earlier checking in a wrong way as compared to old cli. The old cli condition can be found [here](https://github.com/maticnetwork/bor/blob/master/cmd/utils/flags.go#L1576). 
If you see the (add: no-snapshot flag, handle snapshot disable](https://github.com/maticnetwork/bor/pull/323/commits/607645f490486fc70cba38ffae48dfd2d84d1adf) commit, it has the bug fix.